Modules/_sqlite/connection.c

```diff
@@ -1003,13 +1010,15 @@ static PyObject* pysqlite_connection_set_authorizer(pysqlite_Connection* self, P
         return NULL;
     }
 
-    if (PyDict_SetItem(self->function_pinboard, authorizer_cb, Py_None) == -1) {
-        return NULL;
-    }
     rc = sqlite3_set_authorizer(self->db, _authorizer_callback, (void*)authorizer_cb);
     if (rc != SQLITE_OK) {
         PyErr_SetString(pysqlite_OperationalError, "Error setting authorizer callback");
+        Py_INCREF(Py_None);
+        Py_XSETREF(self->function_pinboard_authorizer_cb, Py_None);
```

Comment:
Why are we doing this?

Reply:
I am new this business, so I mimic current behavior like in cursor.c:
```c
Py_INCREF(Py_None);
Py_XSETREF(self->lastrowid, Py_None);
...
Py_DECREF(self->lastrowid);
....
self->lastrowid = _pysqlite_long_from_int64(lastrowid);
```
I will change it to NULL.

Reply:
Fixed.